src/main/java/HttpServer.java

```diff
@@ -7,16 +7,21 @@
 
   public static void main(String[] args) throws IOException {
```

Comment:
That would probably make for a better user experience.

I was also thinking about what we talked about today: IOException comes from the lower-level components, yet it's being bubbled up through their interfaces.